sfms/sfms.py

```diff
@@ -0,0 +1,169 @@
+
+    def add_field(self, name, value):
+        """Add a simple field to the form data."""
+        self.form_fields.append((name, value))
+        return
+
+    def add_file(self, fieldname, filename, mimetype=None):
+        """Add a file to be uploaded."""
+        fileHandle = None
+        try:
+            fileHandle = codecs.open(filename, "rb")
+            body = fileHandle.read()
+        finally:
+            if fileHandle:
+                fileHandle.close()
+        basename = os.path.basename(filename)
+        if mimetype is None:
+            mimetype = mimetypes.guess_type(basename)[0] or 'application/octet-stream'
+        self.files.append((fieldname, basename, mimetype, body))
+        return
+
+    def get_binary(self):
+        """Return a binary buffer containing the form data, including attached files."""
+        part_boundary = '--' + self.boundary
+
+        binary = io.BytesIO()
+        needsCLRF = False
+        # Add the form fields
+        for name, value in self.form_fields:
+            if needsCLRF:
+                binary.write('\r\n')
+            needsCLRF = True
+
+            block = [part_boundary,
+                     'Content-Disposition: form-data; name="%s"' % name,
+                     '',
+                     value
+                     ]
+            binary.write('\r\n'.join(block))
+
+        # Add the files to upload
+        for field_name, filename, content_type, body in self.files:
+            if needsCLRF:
+                binary.write('\r\n')
+            needsCLRF = True
+
+            block = [part_boundary,
+                     str('Content-Disposition: file; name="%s"; filename="%s"' %
+                         (field_name, filename)),
+                     'Content-Type: %s' % content_type,
+                     ''
+                     ]
+            binary.write('\r\n'.join(block))
+            binary.write('\r\n')
+            binary.write(body)
+
+        # add closing boundary marker,
+        binary.write('\r\n--' + self.boundary + '--\r\n')
+        return binary
+
+
+def upload(filename, url, secret):
+    """ Give a filename, URL and secret, post file to the url.
+
+    Based on blogpost: https://blog.thesparktree.com/the-unfortunately-long-story-dealing-with
+    """
+    form = MultiPartForm()
+    form.add_file('file', filename)
+    form_buffer = form.get_binary().getvalue()
+
+    schema, netloc, url, params, query, fragments = urlparse.urlparse(url)
+    http = httplib.HTTPConnection(netloc)
+    http.connect()
+    http.putrequest('POST', url)
+    http.putheader('Secret', secret)
+    http.putheader('Content-type', form.get_content_type())
+    http.putheader('Content-length', str(len(form_buffer)))
+    http.endheaders()
+    http.send(form_buffer)
+    result = http.getresponse()
+
+    print('Status: ' + str(result.status))
+
+
+def main(ini):
+    """ Given some configuration filename, POST all files to the confured URL. """
```

Comment:
```suggestion
    """ Given some configuration filename, POST all files to the configured URL. """
```

Reply:
Don't be making up words Sy.

Reply:
I'm so good an conuring up words though!